lib/pbench/server/api/resources/upload_api.py

```diff
@@ -48,10 +50,15 @@ def __init__(self, config, logger):
             )
         )
 
+    @Auth.token_auth.login_required()
     def put(self, controller):
 
-        # FIXME: This should be assigned from the decoded authorization token
-        username = "pbench"
+        try:
+            user = User.query(id=Auth.token_auth.current_user().id)
+        except Exception:
```

Comment:
So line 53 already handles making a response with a 403 if the token is not valid, and a 401 if it is missing?

Do we always want to report a 500 error here?  Or are there conditions that are not considered an internal error that should be separated out?

Reply:
> So line 53 already handles making a response with a 403 if the token is not valid, and a 401 if it is missing?

Yes if the Authorization header doesn't pass we won't execute the PUT request. However, it returns 401 in both conditions.

> Do we always want to report a 500 error here? Or are there conditions that are not considered an internal error that should be separated out?

Now that you mentioned, I realized we need one more condition to check that is `if user is None`. Other than that if the exception does occur here while querying that means we could not connect to the db or db session is broken or something to do with sqlalchemy db operation and that would be an internal error